quiche/src/lib.rs

```diff
@@ -2008,19 +2017,38 @@ impl Connection {
     ///
     /// [`session()`]: struct.Connection.html#method.session
     #[inline]
+    #[deprecated(note = "Use set_session_and_params()")]
     pub fn set_session(&mut self, session: &[u8]) -> Result<()> {
         let mut b = octets::Octets::with_slice(session);
 
         let session_len = b.get_u64()? as usize;
         let session_bytes = b.get_bytes(session_len)?;
 
-        self.handshake.set_session(session_bytes.as_ref())?;
-
         let raw_params_len = b.get_u64()? as usize;
         let raw_params_bytes = b.get_bytes(raw_params_len)?;
+        self.set_session_and_params((
+            session_bytes.as_ref(),
+            raw_params_bytes.as_ref(),
+        ))
+    }
+
+    /// Configures the given session and transport params for resumption.
+    ///
+    /// On the client, this can be used to offer the given serialized session,
+    /// as returned by [`session_and_params()`], for resumption.
+    ///
+    /// This must only be called immediately after creating a connection, that
+    /// is, before any packet is sent or received.
+    ///
+    /// [`session_and_params()`]: struct.Connection.html#method.session_and_params
+    #[inline]
+    pub fn set_session_and_params(
+        &mut self, session_and_params: (&[u8], &[u8]),
```

Comment:
I used a tuple but I was wondering if a struct would be better to make it easier to understand what is the session and what is the params. WDYT ?

Reply:
@ghedo WDYT ?

Reply:
Why not just separate arguments?

Reply:
I wanted to match it with what is returned by `session_and_param`.